ibis/pandas/aggcontext.py

```diff
@@ -510,12 +513,24 @@ def agg(
                 grouped_frame = indexed_by_ordering
             grouped = grouped_frame[name]
 
-            # perform the per-group rolling operation
-            windowed = self.construct_window(grouped)
-
             if callable(function):
-                window_sizes = windowed.apply(len, raw=True).reset_index(
-                    drop=True
+                # To compute the window_size, we need to contruct a
+                # RollingGroupby and compute count using construct_window.
+                # However, if the RollingGroupby is not numeric, e.g.,
+                # we are calling window UDF on a timestamp column, we
+                # cannot compute rolling count directly because:
+                # (1) windowed.count() will exclude NaN observations
+                #     , which results in incorrect window sizes.
+                # (2) windowed.apply(len, raw=True) will include NaN
+                #     obversations, but doesn't work on non-numeric types.
+                #
+                # To deal with this, we create a _placeholder column
```

Comment:
let's also create an issue on the pandas side (there *might* be one), and then link here

Reply:
Added. (there is an existing one)